yarn-project/noir-contracts/src/contracts/uniswap_contract/src/main.nr

```diff
@@ -0,0 +1,437 @@
+contract Uniswap {
+    use dep::aztec3::abi;
+    use dep::aztec3::abi::PrivateContextInputs;
+    use dep::aztec3::abi::PublicContextInputs;
+    use dep::aztec3::context::PrivateFunctionContext;
+    use dep::aztec3::oracle::public_call;
+    use dep::aztec3::private_call_stack_item::PrivateCallStackItem;
+    use dep::aztec3::public_call_stack_item::PublicCallStackItem;
+    use dep::aztec3::types::point::Point;
+    
+    fn constructor(
+        inputs: pub PrivateContextInputs,
+    ) -> pub [Field; dep::aztec3::abi::PUBLIC_INPUTS_LENGTH] {
+        PrivateFunctionContext::new().finish(inputs)
+    }
+
+    // What we need to make this nicer:
+    // 1. A way to access other (and my own) contract's portal addresses (we need many additional args for no good reason now)
+    // 2. don't pass all args manually to initial context
+    // 3. Variable length arrays for input arguments (8 not enough)
+    // 4. Should have storage for fixed values.
+    // 5. account abstraction using its own "custom" constants instead of using shared constants...
```

Comment:
We should put this list somewhere separate as well, for dogfooding etc. 

Also fine to add that we manually have to set the args array for calling other contracts which is inconvenient when using structs etc.